source/CMakeLists.txt

```diff
@@ -202,8 +208,12 @@ if (BUILD_CPP_IF)
   add_subdirectory (lmp/)
   if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.8)
 #   add_subdirectory (md/)
-    add_subdirectory (ipi/)
-    add_subdirectory (gmx/)
+    if (${OP_CXX_ABI} EQUAL 0 AND USE_TF_PYTHON_LIBS)
```

Comment:
What's the reason to have this line?

Reply:
When the OP_CXX_ABI parameter is set to the ipi project, the following problems are encountered:
```
/usr/bin/ld: CMakeFiles/dp_ipi_low.dir/driver.cc.o: in function `main':
driver.cc:(.text.startup+0x2c9): undefined reference to `XyzFileManager::read(std::string const&, std::vector<std::string, std::allocator<std::string> >&, std::vector<std::vector<double, std::allocator<double> >, std::allocator<std::vector<double, std::allocator<double> > > >&, std::vector<std::vector<double, std::allocator<double> >, std::allocator<std::vector<double, std::allocator<double> > > >&, std::vector<std::vector<double, std::allocator<double> >, std::allocator<std::vector<double, std::allocator<double> > > >&)'
/usr/bin/ld: driver.cc:(.text.startup+0x2f7): undefined reference to `Convert<double>::Convert(std::vector<std::string, std::allocator<std::string> > const&, std::map<std::string, int, std::less<std::string>, std::allocator<std::pair<std::string const, int> > >&)'
collect2: error: ld returned 1 exit status
make[2]: *** [ipi/CMakeFiles/dp_ipi_low.dir/build.make:92: ipi/dp_ipi_low] Error 1
make[1]: *** [CMakeFiles/Makefile2:442: ipi/CMakeFiles/dp_ipi_low.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

Reply:
I think `_GLIBCXX_USE_CXX11_ABI` should be globally set for every library.

Reply:
Before, we never set `_GLIBCXX_USE_CXX11_ABI =1` for the C++ interface, so we don't find the error. But this doesn't mean the error does not exist.